File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const store1 = {}
    for (const member of  participant){
        store1[member] = (store1[member] ?? 0) + 1
    }
    const store2 = {}
    for (const member of  completion){
        store2[member] = (store2[member] ?? 0) + 1
    }
    let i = 0
    for (const user of participant){
        if (store1[user] !== store2[user]){
            return user
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string,int> mp;
    for (string people : participant){
        if (mp.find(people) != mp.end()){
            mp[people] += 1;
        } else {
            mp[people] = 1;
        }
    }
    
    for (string people : completion){
        if (mp.find(people) != mp.end()){
            mp[people] -= 1;
        }
    }
    for (auto [name, count] : mp){
        if (count > 0) return name;
    }
}